### DISCUSSION TITLE
deploy_hook - haproxy

### DISCUSSION THREAD

**Discussion by chriskuku** [acmesh-official/acme.sh#5188] [General]
I need some help to find out how things work together. It happened again, that my website certificate expired, although I have a acme cron file that has this in it:

`50 21 * * * /usr/local/share/acme.sh/acme.sh --cron --home "/var/lib/acme/.acme.sh" > /dev/null`

and in 
`/var/lib/acme/.acme.sh/mydomain.org`
the uptodate certs can be found. Ony `/etc/haproxy/certs/mydomain.org.pem` has not been updated.

I'm wondering where my acme.sh command is resp. to what it is configured.

After update  a deploy_hook should be executed that copies myprivate.key >>fullchain.pem and copies this to
`/etc/haproxy/cers/mydomain.org.pem`. This last step doesn't seem to be executed.

A little help appreciated.

--
Christoph

**Comment by jorgevisentini** [comment]
You can use the command below to deploy to haproxy..
`acme.sh --deploy --deploy-hook haproxy --domain mydomain.org`

It worked for me, but I still don't know if it will be automated, anyway, I have the same question.

**Comment by anttikosk** [comment]
Hi, 
I have exactly same problem, but my certs are updated in the /etc/haproxy/certs folder. 
Only thing is that they are not DEPLOYED to haproxy. I can only deploy them manually. 

So my question is, is acme.sh supposed also to deploy the certs to haproxy?
I have this in acme user crontab:
`42 17 * * * /usr/local/share/acme.sh/acme.sh --cron --home "/var/lib/acme/.acme.sh"  > /dev/null`

Or is that crontab only for renewing the certs?

I really need to know how should the certs be deployed to Haproxy?

**acme.sh v3.0.8 
HAProxy version 2.8.10-1~bpo12+1**
